packages/flutter/test/services/text_input_test.dart

```diff
@@ -2,13 +2,66 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+import 'dart:convert' show utf8;
+
 import 'package:flutter/services.dart';
 import 'package:flutter_test/flutter_test.dart' show TestWidgetsFlutterBinding;
 import '../flutter_test_alternative.dart';
 
 void main() {
   TestWidgetsFlutterBinding.ensureInitialized();
 
+  group('TextInput message channels', () {
+    FakeTextChannel fakeTextChannel;
+    FakeTextInputClient client;
+
+    setUp(() {
+      fakeTextChannel = FakeTextChannel((MethodCall call) async {
+        return;
+      });
+      TextInput.setChannel(fakeTextChannel);
+      client = const FakeTextInputClient();
+    });
+
+    tearDown(() {
+      TextInputConnection.debugResetId();
+      TextInput.setChannel(SystemChannels.textInput);
+    });
+
+    test('text input client handler responds to reattach with setClient', () async {
+      TextInput.attach(client, client.configuration);
+
```

Comment:
Maybe expect here that there has been one method call to ensure that requestExistingInputState only calls setClient once and not twice.

Reply:
Done